internal/handler/flight.go

```diff
@@ -0,0 +1,49 @@
+package handler
+
+import (
+	"context"
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/gin-gonic/gin"
+	"log/slog"
+	"net/http"
+	"strconv"
+)
+
+type FlightService interface {
+	GetAll(ctx context.Context, page, pageSize int, available bool) ([]domain.Flight, error)
+}
+
+const (
+	pageDefault      = 1
+	pageSizeDefault  = 30
+	availableDefault = false
+)
+
+func GetAllFlights(service FlightService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		page, err := strconv.Atoi(c.Query("page"))
```

Comment:
I would suggest having pagination in separate function, because this is a very common usage function. You can paginate pretty much any list. And in case you have a bug in pagination, you would need to fix it only in one place